libcalico-go/lib/validator/v3/validator.go

```diff
@@ -772,6 +779,11 @@ func validateFelixConfigSpec(structLevel validator.StructLevel) {
 				"DeviceRouteSourceAddress", "", reason("is not a valid IPv4 address"), "")
 		}
 	}
+
+	if c.RouteTableRange != nil && c.RouteTableRanges != nil {
+		structLevel.ReportError(reflect.ValueOf(c.RouteTableRange),
+			"RouteTableRange", "", reason("cannot be set when `RouteTableRanges` is also set"), "")
```

Comment:
@caseydavenport is this roughly what you had in mind when you mentioned the api faililng?
Plus corresponding test [here](https://github.com/projectcalico/calico/pull/5432/files#diff-d0fed76674ffad43c00300df4bc3af2c405bf6182041614bd9b5c62b678bc9ccR685)